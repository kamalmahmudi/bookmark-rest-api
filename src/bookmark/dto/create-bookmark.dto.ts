import { IsNotEmpty, IsOptional, IsString } from 'class-validator'

export class CreateBookmarkDto {
  @IsString()
  @IsNotEmpty()
  readonly title: string

  @IsString()
  @IsOptional()
  readonly description?: string

  @IsString()
  @IsNotEmpty()
  readonly link: string
}
