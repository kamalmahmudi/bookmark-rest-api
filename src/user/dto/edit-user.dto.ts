import { IsEmail, IsOptional, IsString } from 'class-validator'

export class EditUserDto {
  @IsEmail()
  @IsOptional()
  readonly email?: string

  @IsString()
  @IsOptional()
  readonly firstName?: string

  @IsString()
  @IsOptional()
  readonly lastName?: string
}
