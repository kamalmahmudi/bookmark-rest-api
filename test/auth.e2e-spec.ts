import { INestApplication } from '@nestjs/common'
import { AuthDto } from 'src/auth/dto'
import { PrismaService } from 'src/prisma/prisma.service'
import * as request from 'supertest'
import { getApp, signupUser } from './jest-utils'

describe('AuthController (e2e)', () => {
  let app: INestApplication
  let prisma: PrismaService
  const dto: AuthDto = {
    email: 'someone@someplace.com',
    password: '123'
  }

  beforeEach(async () => {
    ;[app, prisma] = await getApp()
    await prisma.cleanDb()
  })

  describe('POST /auth/signup', () => {
    // negative testing
    it('should handle missing data', async () => {
      const res = await request(app.getHttpServer())
        .post('/auth/signup')
        .expect(400)
      expect(res.body.message).toMatchSnapshot()
    })

    it('should handle duplicate user', async () => {
      await signupUser(app, dto)
      const res = await request(app.getHttpServer())
        .post('/auth/signup')
        .send(dto)
        .expect(403)
      expect(res.body.message).toBe('Credentials taken')
    })

    // positive testing
    it('should handle correct dto', async () => {
      const res = await request(app.getHttpServer())
        .post('/auth/signup')
        .send(dto)
        .expect(201)
      expect(res)
      expect(res.body.access_token).toBeDefined()
    })
  })

  describe('POST /auth/signin', () => {
    // negative testing
    it('should handle missing data', async () => {
      const res = await request(app.getHttpServer())
        .post('/auth/signin')
        .expect(400)
      expect(res.body.message).toMatchSnapshot()
    })

    it('should handle unknown user', async () => {
      const res = await request(app.getHttpServer())
        .post('/auth/signin')
        .send(dto)
        .expect(403)
      expect(res.body.message).toBe('Credentials incorrect')
    })

    it('should handle incorrect password', async () => {
      await signupUser(app, dto)
      const res = await request(app.getHttpServer())
        .post('/auth/signin')
        .send({
          ...dto,
          password: 'wrong-password'
        })
        .expect(403)
      expect(res.body.message).toBe('Credentials incorrect')
    })

    // positive testing
    it('should handle correct dto', async () => {
      await signupUser(app, dto)
      const res = await request(app.getHttpServer())
        .post('/auth/signin')
        .send(dto)
        .expect(200)
      expect(res.body.access_token).toBeDefined()
    })
  })
})
