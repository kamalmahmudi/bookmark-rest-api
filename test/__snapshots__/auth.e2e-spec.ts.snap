// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthController (e2e) POST /auth/signin negative should handle missing data 1`] = `
Array [
  "email should not be empty",
  "email must be an email",
  "password should not be empty",
  "password must be a string",
]
`;

exports[`AuthController (e2e) POST /auth/signup negative should handle missing data 1`] = `
Array [
  "email should not be empty",
  "email must be an email",
  "password should not be empty",
  "password must be a string",
]
`;
